yt_dlp/extractor/rumble.py

```diff
@@ -236,6 +222,96 @@ def _real_extract(self, url):
         }
 
 
+class RumbleIE(InfoExtractor):
+    _VALID_URL = r'''(?x)
+        https?://(?:www\.)?rumble\.com/
+        (?P<id>v            # ids start with v
+            (?!ideos)       # exclude https://rumble.com/videos
+            [0-9a-zA-Z]{4,} # video page identifier (case insensitive)
+            (?:-[^&?#$/]*)? # usually followed by -<title>.html otherwise redirected
+        )
+        [^/]*$              # queries are allowed, but no slashes
+    '''
+    _EMBED_REGEX = [r'<a class=video-item--a href=(?P<url>/v[0-9a-z.-]+\.html)>']
+    _TESTS = [{
+        'add_ie': ['RumbleEmbed'],
+        'url': 'https://rumble.com/vdmum1-moose-the-dog-helps-girls-dig-a-snow-fort.html',
+        'md5': '53af34098a7f92c4e51cf0bd1c33f009',
+        'info_dict': {
+            'id': 'vb0ofn',
+            'ext': 'mp4',
+            'timestamp': 1612662578,
+            'uploader': 'LovingMontana',
+            'channel': 'LovingMontana',
+            'upload_date': '20210207',
+            'title': 'Winter-loving dog helps girls dig a snow fort ',
+            'description': 'Moose the dog is more than happy to help '
+                           'with digging out this epic snow fort. Great job, Moose!',
+            'channel_url': 'https://rumble.com/c/c-546523',
+            'thumbnail': r're:https://.+\.jpg',
+            'duration': 103,
+            'like_count': int,
+            'view_count': int,
+            'live_status': 'not_live',
+        }
+    }, {
+        'url': 'http://www.rumble.com/vDMUM1?key=value',
+        'only_matching': True,
+    }]
+
+    _WEBPAGE_TESTS = [{
+        'url': 'https://rumble.com/videos?page=2',
+        'playlist_count': 25,
+        'info_dict': {
+            'id': 'videos?page=2',
+            'title': 'All videos',
+            'description': 'Browse videos uploaded to Rumble.com',
+            'age_limit': 0,
+        },
+    }, {
+        'url': 'https://rumble.com/live-videos',
+        'playlist_mincount': 19,
+        'info_dict': {
+            'id': 'live-videos',
+            'title': 'Live Videos',
+            'description': 'Live videos on Rumble.com',
+            'age_limit': 0,
+        },
+    }, {
+        'url': 'https://rumble.com/search/video?q=rumble&sort=views',
+        'playlist_count': 24,
+        'info_dict': {
+            'id': 'video?q=rumble&sort=views',
+            'title': 'Search results for: rumble',
+            'age_limit': 0,
+        },
+    }]
+
+    def _real_extract(self, url):
+        page_id = self._match_id(url)
+        webpage = self._download_webpage(url, page_id)
+
+        rumble_embed = self._downloader.get_info_extractor(RumbleEmbedIE.ie_key())
+        try:
+            url_info = next(rumble_embed.extract_from_webpage(self._downloader, url, webpage))
+        except StopIteration:
+            raise UnsupportedError(url)
+        info = rumble_embed.extract(url_info['url'])
```

Comment:
why not just return url_transparent?

Reply:
I was not sure if it can be used for this purpose. But let's try.
What is the exact behavior? It will add additional metadata keys. But when keys overlap they will be overwritten?

Reply:
When the keys overlap, this extractor's will take precedence (except id)